controllers/fenceagentsremediation_controller_test.go

```diff
@@ -471,16 +477,75 @@ func verifyStatusCondition(far *v1alpha1.FenceAgentsRemediation, nodeName, condi
 			g.Expect(condition).ToNot(BeNil(), "expected condition %v to be set", conditionType)
 			g.Expect(condition.Status).To(Equal(*conditionStatus), "expected condition %v to have status %v", conditionType, *conditionStatus)
 		}
-	}, timeoutDeletion, pollInterval).Should(Succeed())
+	}, timeoutPostRemediation, pollInterval).Should(Succeed())
 }
 
-func verifyRemediationTaintExists(nodeName string, taint *corev1.Taint) {
-	By("Searching for remediation taint")
+// verifyPreRemediationSucceed checks if the remediation CR already has a finazliaer and a remediation taint
+func verifyPreRemediationSucceed(underTestFAR *v1alpha1.FenceAgentsRemediation, nodeName string, taint *corev1.Taint) *v1alpha1.FenceAgentsRemediation {
+	By("Searching for finalizer ")
+	Expect(k8sClient.Get(context.Background(), client.ObjectKey{Name: workerNode, Namespace: defaultNamespace}, underTestFAR)).To(Succeed())
```

Comment:
I think that using global variables to fetch data (e.g. workerNode, underTestFar, etc.) makes it prone to test coupling. It would be better to pass the values as input

Reply:
Good point

Reply:
`workerNode` is a const, but it doesn't change the problem. This function will always look for a node named `workerNode`, which might be correct in this case, but making the request explicit (e.g. `verifyPreRemediationSucceded(targetNode string, ...)` would be clarer IMHO